go/arrow/memory/buffer.go

```diff
@@ -60,15 +61,28 @@ func (b *Buffer) Release() {
 	}
 }
 
+// Reset resets the buffer for reuse.
+func (b *Buffer) Reset(buf []byte) {
```

Comment:
This method leaks memory when the original buffer comes from `b.mem`.

Reply:
@fsaintjacques Should I call `b.mem.Free()` on the original buf?

Reply:
```golang
if buf != b.buf {
    b.mem.Free(buf)
}
b.buf=buf
b.length = len(buf)
```

The special-case is there because one of the use-cases is "resetting" it with the same buf